Tools/LyTestTools/tests/unit/test_o3de_multi_test_framework.py

```diff
@@ -0,0 +1,1184 @@
+                          ')JSON_END'
+
+        results = mock_test_suite._get_results_using_output(
+            mock_log_attribute, mock_test_list, mock_output, mock_log_output)
+        assert len(results) == 3
+        assert 'mock_test_name_pass' in results.keys()
+        assert 'mock_test_name_fail' in results.keys()
+        assert 'mock_test_name_unknown' in results.keys()
+        assert isinstance(results['mock_test_name_pass'], multi_test_framework.Result.Pass)
+        assert isinstance(results['mock_test_name_fail'], multi_test_framework.Result.Fail)
+        assert isinstance(results['mock_test_name_unknown'], multi_test_framework.Result.Unknown)
+
+    @mock.patch('builtins.print')
+    def test_ReportResult_TestPassed_ReportsCorrectly(self, mock_print):
+        mock_log_attribute = 'mock_log_attribute'
+        mock_test_name = 'mock name'
+        mock_test_spec = mock.MagicMock()
+        mock_output = 'mock_output'
+        mock_log_output = 'mock_log_output'
+        mock_pass = ly_test_tools.o3de.multi_test_framework.Result.Pass(
+            mock_log_attribute, mock_test_spec, mock_output, mock_log_output)
+        ly_test_tools.o3de.multi_test_framework.AbstractTestSuite._report_result(mock_test_name, mock_pass)
+        mock_print.assert_called_with(
+            f'Test {mock_test_name}:\nTest Passed\n------------\n'
+            f'|  Output  |\n------------\n{mock_output}\n')
+
+    @mock.patch('pytest.fail')
+    def test_ReportResult_TestFailed_FailsCorrectly(self, mock_pytest_fail):
+        mock_log_attribute = 'mock_log_attribute'
+        mock_test_name = 'mock name'
+        mock_test_spec = mock.MagicMock()
+        mock_output = 'mock_output'
+        mock_log_output = 'mock_log_output'
+        mock_fail = ly_test_tools.o3de.multi_test_framework.Result.Fail(
+            mock_log_attribute, mock_test_spec, mock_output, mock_log_output)
+
+        ly_test_tools.o3de.multi_test_framework.AbstractTestSuite._report_result(mock_test_name, mock_fail)
+        mock_pytest_fail.assert_called_with(
+            f'Test {mock_test_name}:\nTest FAILED\n------------\n|  Output  |\n------------\n{mock_output}\n'
+            f'-------------------------------\n|  {mock_log_attribute} log  |\n-------------------------------\n'
+            '-- No log found --\n')
+
+
+class TestRunningTests(unittest.TestCase):
+
+    @mock.patch('ly_test_tools.o3de.multi_test_framework.AbstractTestSuite._get_results_using_output')
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.retrieve_editor_log_content')
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.retrieve_log_path')
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.get_testcase_module_filepath')
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.cycle_crash_report')
+    @mock.patch('os.path.join', mock.MagicMock())
+    def test_ExecSingleTest_TestSucceeds_ReturnsPass(self, mock_cycle_crash, mock_get_testcase_filepath,
```

Comment:
minor: These heavily mocked tests suggest interface design concerns, though this appears unchanged from the preexisting code.